### ISSUE TITLE
Fully Parse Crunch

### ISSUE THREAD

**Issue by Kixiron** [Kixiron/crunch-lang#6]
Be able to fully parse the grammar spec and extend grammar spec to cover all bases

**Comment by Kixiron** [comment]
- [ ] Doc comments: `::: Comment`
- [ ] Traits:
```crunch
trait TraitName
    fn trait_function()
        println("Hello from TraitName!")
    end
end
```
- [ ] Inline conditionals: `<ret> if <cond> else <ret>`
- [ ] Pattern Matching
- [ ] Enums:
```crunch
enum EnumName
    Variant
    TupleVariant
    OrphanType =>
        orphan_type_field: int
    end
end
```
- [ ] Tuples
- [ ] Vectors
- [ ] Zig-style multiline string literals with `#'` or `#"`:
```crunch
let string = 
    #' Multi
    #'     Line
    #'         String
    #'             Literals

:: Alternatively,
let string = 
    #" Multi
    #"     Line
    #"         String
    #"             Literals
```
- [ ] Decorators (`@decorator`)
  - [ ] `@inline`
  - [ ] `@intrinsic`
  - [ ] `@test`
  - [ ] `@benchmark`
  - [ ] Custom decorators
- [ ] Mandatory Bracing: `a + b * c` is a syntax error, `a + (b * c)` is not

**Comment by Kixiron** [comment]
Variadic Args
- Varargs must be typed in declaration (Types in function signatures are optional)
- Only one vararg per function, and it must be the last declared argument
- Varargs are exposed to the internal function via vector
```
fn print(string: str, args: any*)
```

**Comment by Kixiron** [comment]
Byte strings via `b""` or `b''` that are converted to byte arrays at compile-time
- Requires a static vec type, similar to `Str`

**Comment by Kixiron** [comment]
- [ ] Method access, e.g. `thing.method()`
- [ ] Chained method access, e.g. `thing.method().other_method()`

**Comment by Kixiron** [comment]
- [ ] Break with values, e.g `break true`

**Comment by Kixiron** [comment]
- [ ] Try blocks
```
try
    <body>
catch <err>
    <catch body>
end
```

**Comment by Kixiron** [comment]
The majority of this is solved by 1cfa765

**Comment by Kixiron** [comment]
Almost completed by f8a6c52
